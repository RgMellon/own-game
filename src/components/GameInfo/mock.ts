export default {
  id: '1',
  title: 'Bordelands 3',
  description:
    'Experience the epic space strateggy games that refined the RTS genre',
  price: 215
}
