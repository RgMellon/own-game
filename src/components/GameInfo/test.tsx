import { screen } from '@testing-library/react'
import { rendertWithTheme } from 'utils/tests/helpers'

import GameInfo from '.'

const props = {
  title: 'My Game Tilte',
  description: 'My Game Descriptio',
  price: 200
}

describe('<GameInfo />', () => {
  it('should render game informations', () => {
    rendertWithTheme(<GameInfo {...props} />)

    expect(
      screen.getByRole('heading', { name: /My Game Tilte/i })
    ).toBeInTheDocument()

    expect(screen.getByText('My Game Descriptio')).toBeInTheDocument()
    expect(screen.getByText(/\$200\.00/)).toBeInTheDocument()
  })

  it('should render Buttons', () => {
    rendertWithTheme(<GameInfo {...props} />)

    expect(
      screen.getByRole('button', { name: /add to cart/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('button', { name: /wishList/i })
    ).toBeInTheDocument()
  })
})
