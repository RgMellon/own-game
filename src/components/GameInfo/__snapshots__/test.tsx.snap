// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameInfo /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    GameInfo
  </h1>
</main>
`
